onnxruntime/core/providers/openvino/backend_manager.cc

```diff
@@ -168,9 +168,15 @@ Status BackendManager::ExportCompiledBlobAsEPCtxNode(const onnxruntime::GraphVie
     model_blob_str = model_blob_stream.str();
     ORT_ENFORCE(model_blob_str.size() != 0);
   } else {
-    std::ofstream f(graph_name + ".blob", std::ios::out | std::ios::trunc | std::ios::binary);
+    // Remove extension so we can append suffix to form the complete name of output graph
+    auto blob_name = [&]() {
+      size_t dot = graph_name.find_last_of(".");
+      if (dot == std::string::npos) return graph_name;
+      return graph_name.substr(0, dot);
+    }();
+    std::ofstream f(blob_name + ".blob", std::ios::out | std::ios::trunc | std::ios::binary);
```

Comment:
Define proper name for std::ostream object and handle the exception if the blob_file is being called for I/O operations -
```suggestion
    std::ofstream blob_file(blob_name + ".blob", std::ios::out | std::ios::trunc | std::ios::binary);
    if (!blob_file) { 
     ORT_THROW("Unable to open file for writing."); 
    }

```